src/Pagination.js

```diff
@@ -0,0 +1,240 @@
+'use strict';
+
+import core from 'metal';
+import dom from 'metal-dom';
+import templates from './Pagination.soy';
+import Component from 'metal-component';
+import Soy from 'metal-soy';
+
+class Pagination extends Component  {
+	constructor(value) {
+		super(value);
+
+		this.TOTAL_CONTROLS = 2;
+
+		/**
+		 * 
+		 * @type {Object}
+		 * @default {page: this.page}
+		 */
+		this.lastState = {
```

Comment:
Since this is an internal variable, it should have the `_` suffix: `this.lastState_`.